### ISSUE TITLE
Unable to call function on custom schema

### ISSUE THREAD

**Issue by jsbrain** [supabase/supabase-js#673]
# Bug report

## Describe the bug

When creating a client as in 

```ts
const supabase = createClient<Database, 'myschema'>(
  supabaseUrl,
  supabaseAnonKey,
  {
    db: {
      schema: 'myschema',
    }
  },
);
```

the supabase client will always, despite properly inferring the typed function name for 'myschema', call the function on the `public` schema.

## To Reproduce

Call function on a custom schema:

```ts
await supabase.rpc('function_on_myschema');
```

Result:

```txt
{
    "error": {
        "code": "PGRST202",
        "details": null,
        "hint": "If a new function was created in the database with this name and parameters, try reloading the schema cache.",
        "message": "Could not find the public.function_on_myschema() function or the public.function_on_myschema function with a single unnamed json or jsonb parameter in the schema cache"
    },
    "data": null,
    "count": null,
    "status": 404,
    "statusText": "Not Found"
}
```

## Expected behavior

Call function on the schema supplied to the createClient function.

**Comment by Davepar** [comment]
I think this is "by design." Only `public` tables, views, and functions are available via the API for security. You can always create a wrapper function in the public schema that calls your custom schema function.

**Comment by jsbrain** [comment]
@Davepar if you check the docs you'll realize that this is false. There is an option to specify the schema the client should access as you can see in my first code snippet. This is a feature.

The fact that it infers the types, hence the function name for the passed schema and recommends the function name via itellisense but then tries to execute the function on the public schema instead tells me that this is in fact a bug.

**Comment by Davepar** [comment]
Ah, I learned something new. You're right, you can specify the schema when you create the JS client. However, you need to specifically enable API access for any schema other than the default `public`.

Buried in the docs:

> By default the API server points to the `public` schema. You can enable other database schemas within the Dashboard.
> Go to `Settings > API > Schema` and add the schema which you want to expose to the API.
> 
> Note: each client connection can only access a single schema, so the code above can access the `other_schema` schema but cannot access the `public` schema.

**Comment by soedirgo** [comment]
Hmm, I'm not able to reproduce this. Is this running against a hosted Supabase project? If so, can you file a ticket at https://app.supabase.com/support/new and link this issue?

**Comment by jsbrain** [comment]
It actually seems to be fixed now after installing 2.4.0.

Not sure how the fixes in 2.3.0 or 2.4.0 could have affected this but everything works like a charm now so I guess as long as nobody else experiences this issue I'll just close this.
